rust/src/interpreter/stack.rs

```diff
@@ -0,0 +1,149 @@
+use evmc_vm::{StatusCode, StepStatusCode};
+
+use crate::types::u256;
+
+#[derive(Debug)]
+pub struct Stack(Vec<u256>);
+
+impl Stack {
+    pub fn from(inner: Vec<u256>) -> Self {
+        Self(inner)
+    }
+
+    pub fn empty() -> Self {
+        Self(Vec::new())
+    }
+
+    pub fn into_inner(self) -> Vec<u256> {
+        self.0
+    }
+
+    pub fn push(&mut self, value: impl Into<u256>) -> Result<(), (StepStatusCode, StatusCode)> {
```

Comment:
The Result type is rather verbose, consider introducing an error type.

Reply:
Agreed.